docs/typescript/testing.md

```diff
@@ -2,92 +2,246 @@
 id: testing
 title: Testing TypeScript Workflows
 sidebar_label: Testing
-description: The TypeScript SDK does not yet have a test suite that allows time skipping.
 ---
 
-import CustomWarning from "../components/CustomWarning.js"
+The TypeScript SDK comes with an optional test framework (npm [`@temporalio/testing`](https://www.npmjs.com/package/@temporalio/testing)).
 
-<CustomWarning>
+Upon installation, it will automatically download a test server with time skipping support (more on that later).
 
-Full Testing support is still not available in the TypeScript SDK Beta and will be released before full launch.
+The test framework provides utilities for testing both Activities and Workflows.
 
-</CustomWarning>
+## Testing Activities
 
-The TypeScript SDK does not yet have a test suite that allows time skipping, however, you might be able to get very far without it:
+Activities can be tested with [`MockActivityEnvironment`](https://typescript.temporal.io/api/classes/testing.MockActivityEnvironment)
 
-- Since Activities are async functions, they should be testable as long as you avoid using [Context](https://typescript.temporal.io/api/classes/activity.context) or are able to mock it.
-- You can test Workflows by running them with a [WorkflowClient](https://typescript.temporal.io/api/classes/client.workflowclient).
-- Check [the SDK's own tests](https://github.com/temporalio/sdk-typescript/tree/52f67499860526cd180912797dc3e6d7fa4fc78f/packages/test/src) for more examples.
-- For time skipping, you can implement a multiplier, for example a constant that is `1` in local dev or production, but `0.000001` when testing. Note that `process.env` is unavailable from Worklows so it should be passed in as an argument for the time being.
+The constructor accepts an optional partial activity [`Info`](https://typescript.temporal.io/api/classes/activity.Info) object in case any info fields are needed for the test.
```

Comment:
```suggestion
The constructor accepts an optional partial Activity [`Info`](https://typescript.temporal.io/api/classes/activity.Info) object in case any info fields are needed for the test.
```